src/OrchardCore.Modules/OrchardCore.Media/Drivers/MediaFieldDisplayDriver.cs

```diff
@@ -91,72 +91,71 @@ public override async Task<IDisplayResult> UpdateAsync(MediaField field, IUpdate
         {
             var model = new EditMediaFieldViewModel();
 
-            if (await updater.TryUpdateModelAsync(model, Prefix, f => f.Paths))
-            {
-                // Deserializing an empty string doesn't return an array
-                var items = string.IsNullOrWhiteSpace(model.Paths)
-                    ? []
-                    : JConvert.DeserializeObject<List<EditMediaFieldItemInfo>>(model.Paths, JOptions.CamelCase);
+            await updater.TryUpdateModelAsync(model, Prefix, f => f.Paths);
+
+            // Deserializing an empty string doesn't return an array
+            var items = string.IsNullOrWhiteSpace(model.Paths)
+                ? []
+                : JConvert.DeserializeObject<List<EditMediaFieldItemInfo>>(model.Paths, JOptions.CamelCase);
 
-                // If it's an attached media field editor the files are automatically handled by _attachedMediaFieldFileService.
-                if (string.Equals(context.PartFieldDefinition.Editor(), "Attached", StringComparison.OrdinalIgnoreCase))
+            // If it's an attached media field editor the files are automatically handled by _attachedMediaFieldFileService.
+            if (string.Equals(context.PartFieldDefinition.Editor(), "Attached", StringComparison.OrdinalIgnoreCase))
+            {
+                try
                 {
-                    try
-                    {
-                        field.SetAttachedFileNames(items.Where(i => !i.IsRemoved).Select(i => i.AttachedFileName).ToArray());
-                        await _attachedMediaFieldFileService.HandleFilesOnFieldUpdateAsync(items, context.ContentPart.ContentItem);
-                    }
-                    catch (Exception e)
-                    {
-                        updater.ModelState.AddModelError(Prefix, nameof(model.Paths), S["{0}: There was an error handling the files.", context.PartFieldDefinition.DisplayName()]);
-                        _logger.LogError(e, "Error handling attached media files for field '{Field}'", context.PartFieldDefinition.DisplayName());
-                    }
+                    field.SetAttachedFileNames(items.Where(i => !i.IsRemoved).Select(i => i.AttachedFileName).ToArray());
```

Comment:
Should we update the field itself (which is bound to the content item, which is stored in the db eventually) or should we only update View Model properties such that there is no chance we save the results if that fails. And only update the field if the TryUpdateModeAsync succeeds, like with the previous code.

This is a general question, not just specific to this driver.

Reply:
Always just the viewmodel.

Reply:
Updating the Model instead of the ViewModel just leads into needing to do patching code here and there which is awkward.

Reply:
So we agree. And here it's updating the model, the field, instead of the view model which is created at the top of the method. Should we go through all these cases too or trust content manager to not save the content item because there were errors (pretty sure that's how it works).

Reply:
Yes that's how it works.